tla/consensus/abs.tla

```diff
@@ -0,0 +1,58 @@
+---- MODULE abs ----
+\* Abstract specification for a distributed consensus algorithm.
+\* Assumes that any node can inspect the state of all other nodes. 
+
+EXTENDS Sequences, SequencesExt, Naturals, FiniteSets
+
+CONSTANT Servers, Terms, MaxLogLength, StartTerm
+
+\* Commit logs from each node
+\* Each log is append-only and the logs will never diverge.
+VARIABLE CLogs
+
+
+InitialLogs == {
+    <<>>,
+    <<StartTerm, StartTerm>>,
+    <<StartTerm, StartTerm, StartTerm, StartTerm>>}
+    
+Init ==
+    CLogs \in [Servers -> InitialLogs]
+
+\* A node i can copy a ledger suffix from another node j.
+Copy(i) ==
+    \E j \in Servers : 
+        /\ Len(CLogs[j]) > Len(CLogs[i])
+        /\ \E l \in 1..(Len(CLogs[j]) - Len(CLogs[i])) : 
+                CLogs' = [CLogs EXCEPT ![i] = CLogs[i] \o SubSeq(CLogs[j], Len(CLogs[i]) + 1, Len(CLogs[i]) + l)]
```

Comment:
Substituting `@` for `CLogs[i]` could make it easier to discern `CLogs[j]` from `CLogs[i]`.

Reply:
https://github.com/microsoft/CCF/pull/6438/commits/e9c2fe0af9440f926dee393d34049b5439caf156